instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/ServletOutputStreamInjectionHelper.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet;
+
+import static java.util.logging.Level.WARNING;
+
+import io.opentelemetry.javaagent.bootstrap.servlet.SnippetHolder;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.logging.Logger;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletResponseWrapper;
+
+public class ServletOutputStreamInjectionHelper {
+  private static final Logger logger = Logger.getLogger(HttpServletResponseWrapper.class.getName());
+
+  /**
+   * return true means this method performed the injection, return false means it didn't inject
+   * anything Servlet3OutputStreamWriteAdvice would skip the write method when the return value is
+   * true, and would write the original bytes when the return value is false.
+   */
+  public static boolean handleWrite(
+      byte[] original, int off, int length, InjectionState state, ServletOutputStream out)
+      throws IOException {
+    if (state.isAlreadyInjected() || state.getCharacterEncoding() == null) {
+      return false;
+    }
+    int i;
+    boolean shouldInject = false;
+    for (i = off; i < length && i - off < length; i++) {
+      if (state.processByte(original[i])) {
+        shouldInject = true;
+        break;
+      }
+    }
+    if (!shouldInject) {
+      return false;
+    }
+    state.setAlreadyInjected(); // set before write to avoid recursive loop
+    out.write(original, off, i + 1);
+    try {
+      byte[] snippetBytes = SnippetHolder.getSnippetBytes(state.getCharacterEncoding());
+      out.write(snippetBytes);
+    } catch (UnsupportedEncodingException ignore) {
+      logger.log(WARNING, "UnsupportedEncodingException", ignore);
```

Comment:
let's log all at DEBUG
```suggestion
      logger.log(DEBUG, "UnsupportedEncodingException", ignore);
```